upickle/test/src/upickle/AdvancedTests.scala

```diff
@@ -357,19 +357,27 @@ object AdvancedTests extends TestSuite {
           } yield new String(Array(ch0, ch1, ch2, ch3, ch4, ch5, ch6, ch7))
         }
 
-        val jsonString =
+        def jsonString(elements: Int) =
           zeroHashCodeStrings
             .map(s => ujson.write(s))
-            .take(1000000)
+            .take(elements)
             .mkString("{", s":null,", ":null}")
 
-        // This is expected to have sub-linear behavior, due to use of `LinkedHashMap` in `ujson.Value`
-        // which is not robust against collisions
-        //        ujson.read(jsonString)
-
-        // Make sure this can pass in Scala 3. We do not use `ujson.Value` in this code path at all,
-        // so it should finish in a reasonable amount of time and without issue
-        upickle.default.read[Foo416](jsonString)
+        test("issue-416") {
+          upickle.default.read[Foo416](jsonString(1000000))
+          ()
+        }
+        test("issue-446") {
+          sys.props("java.vm.name") match {
+            case "Scala.js" | "Scala Native" =>
+            // The fix assumes a hash collision safe java.util.LinkedHashMap
+            // implementation. When/if other platforms will have such characteristics
+            // ujson doesn't need to change.
+            case _ =>
+              ujson.read(jsonString(100000))
+          }
+          ()
+        }
```

Comment:
In what way will this test catch the issue?

Reply:
Using the previous implementation this was not completing. I'm not sure that if I waited more, it would actually complete, but since it's a performance problem I don't know how to test it. Putting time constraints leads to flaky tests, so I would avoid them.